components/ChargeBar/ChargeBar.jsx

```diff
@@ -0,0 +1,151 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import styled from 'styled-components';
+import {
+  components,
+  spacing as themeSpacing,
+  baseFontSize as defaultBaseFontSize,
+} from '../shared/theme';
+import { hexToRgba } from '../shared';
+import Icon from '../Icon';
+
+const Bar = styled.div`
+  border-radius: 4px;
+  overflow: hidden;
+  position: relative;
+
+  ${({ width, background, spacing: { xxsmall } }) => `
+    background-color: ${hexToRgba(background, 0.39)};
+    height: 4px;
+    width: ${width}; 
+    margin-top: ${xxsmall}px;
+  `}
+`;
+
+const Content = styled.div`
+  border-radius: 8px;
+  display: inline-block;
+  height: 100%;
+  left: 0;
+  position: absolute;
+  top: 0;
+  transition: width 0.4s linear;
+
+  ${({ background, progress }) => `
+    background-color: ${background};
+    width: ${progress}%;
+  `}
+`;
+
+const Wrapper = styled.div`
+  display: flex;
+  flex-direction: column;
+  font-stretch: normal;
+  font-style: normal;
+  font-weight: normal;
+  letter-spacing: 0.18px;
+  line-height: 1.5;
+`;
+
+const ProgressLabel = styled.span`
+  display: inline-block;
+`;
+
+const InfoBlock = styled.div`
+  display: flex;
+  justify-content: space-between;
+  ${({ width, background, baseFontSize }) => `
+    width: ${width};
+    color: ${background};
+    font-size: ${baseFontSize * 0.875}px;
+  `}
+`;
+
+const TextBlock = styled.span`
+  display: flex;
+  align-items: center;
+`;
+
+const LabelInfo = styled.span`
+  ${({ spacing: { xsmall } }) => `
+    margin-left: ${xsmall}px;
+  `}
+  height: 17px;
+`;
+
+ProgressLabel.displayName = 'ProgressLabel';
+LabelInfo.displayName = 'LabelInfo';
+
+const ChargeBar = props => {
+  const {
+    width,
+    progressPercent,
+    label,
+    skin,
+    theme: {
+      components: {
+        chargeBar: {
+          skins: {
+            [skin]: { background },
+          },
+        },
+      },
+      spacing,
+      baseFontSize,
+    },
+  } = props;
+
+  return (
+    <Wrapper>
+      <InfoBlock
+        width={width}
+        background={background}
+        baseFontSize={baseFontSize}
+      >
+        <TextBlock>
+          <Icon name="whatshot" size="small" skin={background} />
+          <LabelInfo spacing={spacing}>{label}</LabelInfo>
+        </TextBlock>
+        <ProgressLabel>{progressPercent}%</ProgressLabel>
+      </InfoBlock>
+      <Bar
+        background={background}
+        width={width}
+        spacing={spacing}
+        role="progressbar"
+      >
+        <Content background={background} progress={progressPercent} />
+      </Bar>
+    </Wrapper>
+  );
+};
```

Comment:
Man, I guess there are many complexity and too many elements in this simple component. I think width can be controlled by wrapper, and text elements can be wrapped by only one element. I also think span elements don't need to be styled component.

```suggestion
const InfoBlock = styled.div`
  display: flex;
  justify-content: space-between;
  
  ${({ spacing: { xsmall } }) => css`
    > * + * {
      margin-left: ${xsmall}px;
    }
  `}

  svg + span {
    flex: 1;
  }
`;

const ChargeBar = props => {
  const {
    width,
    progressPercent,
    label,
    skin,
    theme: {
      components: {
        chargeBar: {
          skins: {
            [skin]: { background },
          },
        },
      },
      spacing,
      baseFontSize,
    },
  } = props;

  return (
    <Wrapper width={width} background={background} baseFontSize={baseFontSize}>
      <InfoBlock spacing={spacing}>
        <Icon name="whatshot" size="small" skin={background} />
        <span>{label}</span>
        <span>{progressPercent}%</span>
      </InfoBlock>
      <Bar
        background={background}
        progress={progressPercent}
        spacing={spacing}
        role="progressbar"
      />
    </Wrapper>
  );
};
```

Reply:
Good tip! Ia agree that have too many elements. I did remove it.
But i prefere to keep some elements in styled, in intention of make code clearer.

See what i did and say what did you think